tests/database/test_database.py

```diff
@@ -0,0 +1,150 @@
+        password_hash = access_db.get_user_info(user_id=1, field="password_hash")
+        # password would be hashed in the actual database
+        # it is not in tests for ease of use
+        assert password_hash == "hashed_password"
+
+    def test_get_username_success(self, access_db, create_user):
+        """Test that the correct username is retrieved for an existing user."""
+        username = access_db.get_user_info(user_id=1, field="username")
+        assert username == "testuser"
+
+    def test_get_email_success(self, access_db, create_user):
+        """Test that the correct email is retrieved for an existing user."""
+        email = access_db.get_user_info(user_id=1, field="email")
+        assert email == "testuser@example.com"
+
+    def test_get_date_created_success(self, access_db, create_user):
+        """Test that the correct creation date is retrieved for an existing user."""
+        created_at = access_db.get_user_info(user_id=1, field="created_at")
+        # Checks a datetime is returned
+        assert isinstance(created_at, datetime)
+
+    def test_get_user_not_found(self, access_db):
+        """Test that a GetUserFromIdError is raised when the user does not exist."""
+        with pytest.raises(GetUserFromIdError, match="User with ID 1 not found"):
+            access_db.get_user_info(user_id=1, field="password_hash")
+
+    def test_get_user_invalid_field(self, access_db, create_user):
+        """Test that a ValueError is raised for an invalid field."""
+        with pytest.raises(ValueError, match="Invalid field: invalid_field"):
+            access_db.get_user_info(user_id=1, field="invalid_field")
+
+    def test_get_user_no_field(self, access_db, create_user):
+        """Test that the correct user object is returned when field=None."""
+        user = access_db.get_user_info(user_id=1, field=None)
+        # Ensure the returned user object has the expected properties
+        assert user.id == 1
+        assert user.username == "testuser"
+        assert user.email == "testuser@example.com"
+
+
+class TestUpdateEmail:
+    """Test updating user's email."""
+
+    def test_update_email_success(self, access_db, create_user):
+        """Test that the email is updated successfully."""
+
+        new_email = "new_email@example.com"
+
+        # Update the email for the user
+        access_db.update_email(user_id=1, new_email=new_email)
+
+        # Verify that the email was updated
+        updated_email = access_db.get_user_info(user_id=1, field="email")
+        print(updated_email)
+        assert updated_email == new_email
+
+    def test_update_email_user_not_found(self, access_db):
+        """Test that a GetUserFromIdError is raised when the user does not exist."""
+
+        new_email = "new_email@example.com"
+
+        # Try to update the email of a non-existing user
+        with pytest.raises(GetUserFromIdError):
+            access_db.update_email(user_id=1, new_email=new_email)
```

Comment:
```suggestion
    def test_update_email_user_not_found(self, access_db, create_user):
        """Test that a GetUserFromIdError is raised when the user does not exist."""

        new_email = "new_email@example.com"

        # Try to update the email of a non-existing user
        with pytest.raises(GetUserFromIdError):
            access_db.update_email(user_id=2, new_email=new_email)
```
same as earlier comment

Reply:
Resolved in [d91ce60](https://github.com/rowan04/L5-SE-CW2-QualityPlan/pull/14/commits/d91ce60cb4fba350a53d8e52a5b499b8a095c5d1)